src/main/java/gregtech/common/metatileentities/multi/electric/generator/FueledMultiblockController.java

```diff
@@ -98,10 +93,19 @@ protected boolean shouldUpdate(MTETrait trait) {
     }
 
     @Override
-    protected boolean checkStructureComponents(List<IMultiblockPart> parts, Map<MultiblockAbility<Object>, List<Object>> abilities) {
-        //noinspection SuspiciousMethodCalls
-        return abilities.containsKey(MultiblockAbility.IMPORT_FLUIDS) &&
-                abilities.containsKey(MultiblockAbility.OUTPUT_ENERGY);
+    public TraceabilityPredicate autoAbilities() {
+        return autoAbilities(true, true, true);
+    }
+
+    public TraceabilityPredicate autoAbilities(boolean checkEnergyOut,
+                                               boolean checkMaintainer,
+                                               boolean checkFluidIn) {
+        TraceabilityPredicate predicate = !checkEnergyOut ? new TraceabilityPredicate() : abilities(MultiblockAbility.OUTPUT_ENERGY);
+        predicate = predicate.or(!checkMaintainer ? null : super.autoAbilities());
+        if (checkFluidIn) {
+            predicate = predicate.or(abilities(MultiblockAbility.IMPORT_FLUIDS).setMinGlobalLimited(1).setPreviewCount(1));
+        }
+        return predicate;
```

Comment:
This also needs to check for Fluid Outputs, as some multiblock generators, such as the Large Turbines, will output fluid upon completing a recipe. Right now, the structures will not form with a fluid output.

Reply:
yeah